frontend/src/app/pages/category-list/category-list.component.ts

```diff
@@ -175,10 +206,17 @@ export class CategoryListComponent implements OnInit, OnDestroy {
       .pipe(
         takeUntil(this.unsubscribe)
       )
-      .subscribe(
-        (data) => {
+      .subscribe({
+        next: (data) => {
+          this.loading = true;
```

Comment:
`this.loading = true` in `subscripe.next` section is wrong, because it will be executed after api call. `this.loading = true` must be moved before subscribe method call (`this.categoryService.getAllCategories()`), unless this is modal window click event (`this.dialog.open(DialogWindowComponent).afterClosed().subscribe({`)